website/versioned_docs/version-v1.x/spec.md

```diff
@@ -0,0 +1,944 @@
+---
+id: spec
+title: Spec
+
+---
+
+# MACI 1.0 Specification (for audit)
```

Comment:
A lot of this content was already spread between the different markdown files (version-v1.x), for instance you will find all of the concepts around keypairs, messages and hash functions inside the primitives.md file. 
As not everything was added, I believe it might be more beneficial to actually extract the content that we are missing that is useful, and place it in the appropriate doc file (contracts, primitives, circuits, etc.).

Reply:
MACI 1.0 Specification is a comprehensive document created specifically for auditors and those who need an in-depth understanding of MACI's technical aspects. While it shares topics like keypairs and hash functions with existing documentation, the specification delves into greater depth and provides an interconnected context that the current documents do not fully capture. This comprehensive approach is similar to how the [Ethereum Yellow Paper](https://ethereum.github.io/yellowpaper/paper.pdf) and the [Go language specification](https://go.dev/ref/spec) are provided as standalone documents, serving as in-depth references for their respective technologies.

Given these points, I believe keeping the MACI 1.0 Specification as a standalone document on our site would greatly benefit users needing detailed technical info.

Looking forward to hearing your thoughts on this.

Reply:
I thought that the documentation itself would "greatly benefit users needing detailed technical info." and having everything divided into each component (which if you read you'll find contain extracts from this spec) might be easier to navigate than one standalone document.
Anyways please remove references to the audit (like the first section) if you want to use this as protocol spec

Reply:
I tend to agree with @ctrlc03. I fear this duplicates some content, which risks future mistakes of us updating content in one place but not the other. But if the goal is to serve as a static specification (like the Ethereum Yellow Paper), I'm ok with having it.

Reply:
If we want to serve it as a static specification, how do we host it in a way that avoids confusing the user? as mentioned, some information in this document is outdated, do we go and change it (and basically let it be a plain copy of what we already have in all separate docs, which already shared quite a lot of content with), or leave it as it is (minus the references/wording of the audit) and specify clearly (somewhere/somehow) that it is outdated? Maybe we host it as a static file and just add a link to it in the introduction.md doc (same way as we reference to the original etherresearch post)?

Reply:
@ctrlc03 I'd like to challenge that this documentation is outdated. Have there been any breaking changes since the last audit? For example, have there been changes to how the circuit works, or to the cryptographic primitives like the Baby Jubjub curve or Poseidon? Have there been modifications in the way we use the accumulator queue, or in the structure of the verifying key? Until such core components are changed, this is the only complete reference for understanding the technical details (!= just implementation details) of MACI

Reply:
I don't think there is a duplication; rather, our working documentation is currently incomplete. For example, the [primitives](https://maci.pse.dev/docs/primitives) page lacks some important details that I attached below. width="1325" alt="Screenshot 2023-11-30 at 07 46 09" src="https://github.com/privacy-scaling-explorations/maci/assets/1610146/962f3dc1-1c85-40a4-a2d4-73d06a1ae1e6">

Reply:
Oh yea no one said all of this is outdated, hence why I used the word "some" (some information in this document is outdated). 
Also: "I believe it might be more beneficial to actually extract the content that we are missing that is useful, and place it in the appropriate doc file (contracts, primitives, circuits, etc.)." -> as you also highlighted as an example, the primitives file is missing some information present on this, it would have been very neat to add it in the respective sections 🙂 
Looks like we agreed to keep it as its own entry within the v1 docs, my only request is that you remove this section: "## Audit scope" and change the wording of the "### Statements that we wish to challenge" section -> as this is supposed to be used as a spec, it should be worded as such and not as a document addressed to an audit firm

Reply:
@ctrlc03 thank you for the clarification :) However, I'd like to preserve the original form since it's a primary resource (fyi [this link](https://www.wgu.edu/blog/what-difference-between-primary-secondary-source2304.html#openSubscriberModal) explains the difference between primary and secondary sources). Also, I believe in maintaining the original context as much as possible to prevent any loss of understanding about the data's background and origin due to middleman intervention. Separately, the "Audit scope" section allows readers to know precisely exactly which commit the audit was based on (it's not stated any other places), indicating that the document reflects that particular time point. Through this, discerning readers can infer that, although the current codebase may have changes since the audited version, they can still expect to find relevant information on the core design, as both the document and the current codebase pertain to version v1.